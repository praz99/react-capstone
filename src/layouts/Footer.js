import React from 'react';
import '../styles/Footer.css';

const Footer = () => (
  <div className="footer">
    <div className="footer-left">
      <p>
        Developed by
        <span className="name"> Prajwal Thapa</span>
      </p>
    </div>
    <div className="footer-right">
      <a className="footer-icons" aria-label="github link" href="https://github.com/praz99" target="_blank" rel="noreferrer"><i className="fab fa-github" /></a>
      <a className="footer-icons" aria-label="linked-in link" href="https://linkedin.com/in/prazwal-thapa" target="_blank" rel="noreferrer"><i className="fab fa-linkedin-in" /></a>
      <a className="footer-icons" aria-label="twitter link" href="https://twitter.com/thapa_praz" target="_blank" rel="noreferrer"><i className="fab fa-twitter" /></a>
    </div>
  </div>
);

export default Footer;
