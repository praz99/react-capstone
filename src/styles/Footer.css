.footer {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
  background-color: lightblue;
  height:  50px;
  display: flex;
  align-items: center;
  padding: 5px 10px;
}

.footer-left {
  width: 50%;
  font-size: 12px;
}

.name {
  color: #b56988;
  font-size: 15px;
  font-weight: normal;
}

.footer-right {
  position: absolute;
  right: 20px;
}

.footer-icons {
  color: black;
  padding: 0 10px;
  font-size: 20px;
}