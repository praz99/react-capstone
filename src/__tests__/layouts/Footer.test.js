import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Footer from '../../layouts/Footer';

afterEach(cleanup);

it('renders Footer component correctly', () => {
  const footer = renderer.create(<Footer />).toJSON();
  expect(footer).toMatchSnapshot();
});

it('should display developer name', () => {
  render(<Footer />);
  const devName = screen.getByText(/Developed by/);
  expect(devName).toBeInTheDocument();
});

it('should display link to linked-in page', () => {
  render(<Footer />);
  const linkInLink = screen.getByLabelText('linked-in link');
  expect(linkInLink).toBeInTheDocument();
});

it('should display link to github page', () => {
  render(<Footer />);
  const githubLink = screen.getByLabelText('github link');
  expect(githubLink).toBeInTheDocument();
});

it('should display link to twitter page', () => {
  render(<Footer />);
  const twitterLink = screen.getByLabelText('twitter link');
  expect(twitterLink).toBeInTheDocument();
});
