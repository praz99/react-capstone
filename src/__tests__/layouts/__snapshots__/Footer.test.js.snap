// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component correctly 1`] = `
<div
  className="footer"
>
  <div
    className="footer-left"
  >
    <p>
      Developed by
      <span
        className="name"
      >
         Prajwal Thapa
      </span>
    </p>
  </div>
  <div
    className="footer-right"
  >
    <a
      aria-label="github link"
      className="footer-icons"
      href="https://github.com/praz99"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github"
      />
    </a>
    <a
      aria-label="linked-in link"
      className="footer-icons"
      href="https://linkedin.com/in/prazwal-thapa"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-linkedin-in"
      />
    </a>
    <a
      aria-label="twitter link"
      className="footer-icons"
      href="https://twitter.com/thapa_praz"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-twitter"
      />
    </a>
  </div>
</div>
`;
